plugins/languages/rust/rustaceanvim/default.nix

```diff
@@ -58,7 +58,16 @@ helpers.neovim-plugin.mkNeovimPlugin config {
       '';
     in
     mkMerge [
-      { extraPackages = [ cfg.rustAnalyzerPackage ]; }
+      {
+        extraPackages = [ cfg.rustAnalyzerPackage ];
+        # TODO: remove after 24.11
+        warnings = optional (cfg.settings ? server && cfg.settings.server ? settings) ''
```

Comment:
Could use `hasAttrByPath` for deeply nested attrs. This is fine though since it's only two checks.